pkg/segment/writer/segmetarw.go

```diff
@@ -108,27 +108,31 @@ func ReadLocalSegmeta(readFullMeta bool) []*structs.SegMeta {
 		return retVal
 	}
 
+	workSfm := &structs.SegFullMeta{}
+
 	// continue reading/merging from individual segfiles
 	for _, smentry := range retVal {
-		sfmData, _ := readSfm(smentry.SegmentKey)
-		if sfmData == nil {
+		err := readSfm(smentry.SegmentKey, workSfm)
+		if err != nil {
+			// error is logged in the func
 			continue
 		}
+
 		if smentry.AllPQIDs == nil {
-			smentry.AllPQIDs = sfmData.AllPQIDs
+			smentry.AllPQIDs = workSfm.AllPQIDs
 		} else {
-			utils.MergeMapsRetainingFirst(smentry.AllPQIDs, sfmData.AllPQIDs)
+			utils.MergeMapsRetainingFirst(smentry.AllPQIDs, workSfm.AllPQIDs)
 		}
 		if smentry.ColumnNames == nil {
-			smentry.ColumnNames = sfmData.ColumnNames
+			smentry.ColumnNames = workSfm.ColumnNames
 		} else {
-			utils.MergeMapsRetainingFirst(smentry.ColumnNames, sfmData.ColumnNames)
+			utils.MergeMapsRetainingFirst(smentry.ColumnNames, workSfm.ColumnNames)
 		}
 	}
 	return retVal
 }
 
-func readSfm(segkey string) (*structs.SegFullMeta, error) {
+func readSfm(segkey string, sfm *structs.SegFullMeta) error {
 
 	sfmFname := getSegFullMetaFnameFromSegkey(segkey)
```

Comment:
nit: maybe check if `sfm == nil`. I think your code still works because the unmarshalling will return an error in that case (see https://go.dev/play/p/7pxm6eble-j)

Reply:
marshalling is going to return error if sfm is nil. So I think I will skip checking for nil.